x-pack/plugins/infra/public/components/asset_details/hooks/use_metadata.ts

```diff
@@ -44,7 +45,7 @@ export function useMetadata(
     (async () => {
       await makeRequest();
     })();
-  }, [makeRequest]);
+  }, [makeRequest, refetch]);
```

Comment:
Q: if you expose `makeRequest` in the `return`, as in the [`use_snapshot`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/pages/metrics/inventory_view/hooks/use_snaphot.ts#L95), I think you would be able to propagate it to the `MetadataErrorCallout` and use it in the `onClick` event directly. Do you think that would be better than controlling the `refetch` flag?

Reply:
That's a good idea, thank you! I exposed directly `makeRequest` as suggested and it works fine :)